src/MUIDataTable.js

```diff
@@ -517,7 +526,10 @@ class MUIDataTable extends React.Component {
       const caseSensitive = this.options.caseSensitive;
       const filterType = column.filterType || this.options.filterType;
       if (filterVal.length) {
-        if (filterType === 'textField' && !this.hasSearchText(columnVal, filterVal, caseSensitive)) {
+        if (column.filterOptions && column.filterOptions.logic) {
+          if (!Array.isArray(column.filterOptions.names)) console.error('filterOptions with logic requires names');
```

Comment:
Actually, I was thinking that while this functionality is being added, we should allow usage of it even when there are no custom filter names. People would then still be able to provide custom filters for existing filter options.

Reply:
I don't understand want you want here.

Reply:
I don't want to require use of `names` with `logic`, as I want people to be able to add `logic` without `names`. So I think we should remove this line entirely.